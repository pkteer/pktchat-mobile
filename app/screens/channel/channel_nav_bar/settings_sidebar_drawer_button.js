// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import PropTypes from 'prop-types';
import React, {PureComponent} from 'react';
import {intlShape} from 'react-intl';
import {
    TouchableOpacity,
    View,
} from 'react-native';
import {connect} from 'react-redux';

import CompassIcon from '@components/compass_icon';
import {getTheme} from '@mm-redux/selectors/entities/preferences';
import {t} from '@utils/i18n';
import {preventDoubleTap} from '@utils/tap';
import {makeStyleSheetFromTheme} from '@utils/theme';

export class SettingsSidebarDrawerButton extends PureComponent {
    static propTypes = {
        openSidebar: PropTypes.func.isRequired,
        theme: PropTypes.object,
    };

    static defaultProps = {
        theme: {},
    };

    static contextTypes = {
        intl: intlShape.isRequired,
    };

    handlePress = preventDoubleTap(() => {
        this.props.openSidebar();
    });

    render() {
        const {theme} = this.props;

        const {formatMessage} = this.context.intl;

        const buttonDescriptor = {
            id: t('navbar.more_options.button'),
            defaultMessage: 'More Options',
            description: 'Accessibility helper for more options button in channel header.',
        };
        const accessibilityLabel = formatMessage(buttonDescriptor);

        const buttonHint = {
            id: t('navbar.more_options.hint'),
            defaultMessage: 'Opens the more options right hand sidebar',
            description: 'Accessibility helper for explaining what the more options button in the channel header will do.',
        };
        const accessibilityHint = formatMessage(buttonHint);

        const style = getStyleFromTheme(theme);

        const icon = (
            <CompassIcon
                name='dots-vertical'
                size={25}
                color={theme.sidebarHeaderTextColor}
            />
        );

        return (
            <TouchableOpacity
                accessible={true}
                accessibilityHint={accessibilityHint}
                accessibilityLabel={accessibilityLabel}
                accessibilityRole='button'
                testID='settings_sidebar_drawer.button'
                onPress={this.handlePress}
                style={style.container}
            >
                <View style={style.wrapper}>
                    {icon}
                </View>
            </TouchableOpacity>
        );
    }
}

const getStyleFromTheme = makeStyleSheetFromTheme((theme) => {
    return {
        container: {
            width: 44,
        },
        wrapper: {
            alignItems: 'center',
            flex: 1,
            flexDirection: 'column',
            justifyContent: 'center',
            marginLeft: 8,
        },
        mention: {
            color: theme.mentionColor,
            fontSize: 10,
        },
    };
});

function mapStateToProps(state) {
    return {
        theme: getTheme(state),
    };
}

export default connect(mapStateToProps)(SettingsSidebarDrawerButton);
