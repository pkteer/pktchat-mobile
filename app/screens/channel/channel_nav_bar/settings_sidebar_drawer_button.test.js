// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.

import React from 'react';

import Preferences from '@mm-redux/constants/preferences';
import {shallowWithIntl} from '@test/intl-test-helper';

import {SettingsSidebarDrawerButton} from './settings_sidebar_drawer_button';

describe('SettingsSidebarDrawerButton', () => {
    const baseProps = {
        openSidebar: jest.fn(),
        theme: Preferences.THEMES.denim,
    };

    test('should match, full snapshot', () => {
        const wrapper = shallowWithIntl(
            <SettingsSidebarDrawerButton {...baseProps}/>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    test('Should be accessible', () => {
        const wrapper = shallowWithIntl(
            <SettingsSidebarDrawerButton {...baseProps}/>,
        );
        expect(wrapper.props().accessible).toBeTruthy();
    });

    test('Should have the correct accessibilityHint', () => {
        const wrapper = shallowWithIntl(
            <SettingsSidebarDrawerButton {...baseProps}/>,
        );
        expect(wrapper.props().accessibilityHint).toEqual('Opens the more options right hand sidebar');
    });

    test('Should have the correct accessibilityLabel', () => {
        const wrapper = shallowWithIntl(
            <SettingsSidebarDrawerButton {...baseProps}/>,
        );
        expect(wrapper.props().accessibilityLabel).toEqual('More Options');
    });
});
