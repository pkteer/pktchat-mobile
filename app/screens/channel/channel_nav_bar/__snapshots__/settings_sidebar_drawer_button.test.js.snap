// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsSidebarDrawerButton should match, full snapshot 1`] = `
<TouchableOpacity
  accessibilityHint="Opens the more options right hand sidebar"
  accessibilityLabel="More Options"
  accessibilityRole="button"
  accessible={true}
  onPress={[Function]}
  style={
    Object {
      "width": 44,
    }
  }
  testID="settings_sidebar_drawer.button"
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "marginLeft": 8,
      }
    }
  >
    <CompassIcon
      color="#ffffff"
      name="dots-vertical"
      size={25}
    />
  </View>
</TouchableOpacity>
`;
